tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,174 @@
+    /* No local chunk suitable for this data has been created yet,
+     * hence chunk should be NULL.
+     */
+    chunk = flb_chunk_get(store, key1);
+    TEST_CHECK(chunk == NULL);
+    ret = flb_buffer_put(store, chunk, key1, data, data_len);
+    TEST_CHECK(ret == 0);
+
+    /* Ensure that the data stored and retrieved from the chunk is the same. */
+    chunk = flb_chunk_get(store, key1);
+    check_chunk(chunk, key1, data);
+
+    ret = flb_remove_chunk_files(chunk);
+    TEST_CHECK(ret == 0);
+    flb_chunk_destroy(chunk);
+    flb_free(out);
+    flb_free(store);
+}
+
+static void test_flb_buffer_put_valid_chunk()
+{
+    int ret;
+    struct flb_local_buffer *store = NULL;
+    struct flb_output_instance *out = NULL;
+    
+    store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    TEST_CHECK(store != NULL);
+    out = flb_calloc(1, sizeof(struct flb_output_instance));
+    TEST_CHECK(out != NULL);
+    
+    store->dir = BUFFER_DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    store->ins = out;
+    mk_list_init(&store->chunks);
+    TEST_CHECK(mk_list_size(&store->chunks) == 0);
+
+    char *data = TEST_DATA;
+    char *key2 = KEY_2;
+    size_t data_len = strlen(data);
+    struct flb_local_chunk *chunk;
+
+    /* No local chunk suitable for this data has been created yet,
+     * hence chunk should be NULL.
+     */
+    chunk = flb_chunk_get(store, key2);
+    TEST_CHECK(chunk == NULL);
+
+    ret = flb_buffer_put(store, chunk, key2, data, data_len);
+    TEST_CHECK(ret == 0);
+
+    /* A new chunk associated with key2 was created in the above statement,
+     * hence this time, chunk should not be NULL.
+     */
+    chunk = flb_chunk_get(store,key2);
+    TEST_CHECK(chunk != NULL);
+
+    chunk = flb_chunk_get(store, key2);
+    check_chunk(chunk, key2, data);
+
+    ret = flb_remove_chunk_files(chunk);
+    TEST_CHECK(ret == 0);
+    flb_chunk_destroy(chunk);
+    flb_free(out);
+    flb_free(store);
+}
+
+static void test_flb_init_local_buffer()
+{
+    int ret;
+    char *data = TEST_DATA;
+    char *key3 = KEY_3;
+    struct flb_local_chunk *chunk;
+    struct flb_local_buffer *store = NULL;
+    struct flb_local_buffer *new_store = NULL; 
+    struct flb_output_instance *out = NULL;
+    
+    store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    TEST_CHECK(store != NULL);
+    new_store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    TEST_CHECK(new_store != NULL);
+    out = flb_calloc(1, sizeof(struct flb_output_instance));
+    TEST_CHECK(out != NULL);
+    
+    store->dir = BUFFER_DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    store->ins = out;
+    mk_list_init(&store->chunks);
+    TEST_CHECK(mk_list_size(&store->chunks) == 0);
+
+    new_store->dir = BUFFER_DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    new_store->ins = out;
+    mk_list_init(&new_store->chunks);
+    TEST_CHECK(mk_list_size(&new_store->chunks) == 0);
```

Comment:
in all of these tests, do the check of mk_list_size after you add a chunk or call init

Reply:
Done